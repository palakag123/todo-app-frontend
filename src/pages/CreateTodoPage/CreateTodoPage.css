.create-todo-page-container{
    display: flex;
}
@media only screen and (max-width: 770px) {
    .create-todo-page-container{
        display: flex;
        flex-direction: column;
    }
}