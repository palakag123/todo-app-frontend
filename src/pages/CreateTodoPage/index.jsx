import React from "react";
import PropTypes from "prop-types";
import CreateTodo from "../../components/CreateTodo";
import UserDetails from "../../components/UserDetails";
import "./CreateTodoPage.css";

function CreateTodopage({ todos, setTodos }) {
  return (
    <div className="create-todo-page-container">
      <UserDetails />
      <CreateTodo todos={todos} setTodos={setTodos} />
    </div>
  );
}
CreateTodopage.propTypes = {
  todos: PropTypes.arrayOf(
    PropTypes.shape({
      id: PropTypes.number.isRequired,
      todo: PropTypes.string.isRequired,
      // eslint-disable-next-line comma-dangle
    })
  ).isRequired,
  setTodos: PropTypes.func.isRequired,
};
export default CreateTodopage;
