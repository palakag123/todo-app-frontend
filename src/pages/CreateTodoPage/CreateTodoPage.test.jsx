/* eslint-disable comma-dangle */
import React from "react";
import { render, screen } from "@testing-library/react";
import CreateTodoPage from "./index";

const mockNavigate = jest.fn();
jest.mock("react-router-dom", () => ({ useNavigate: () => mockNavigate }));

describe("CreateTodoPage", () => {
  const mockTodos = [
    { id: 1, todo: "first todo list item" },
    { id: 2, todo: "second todo list item" },
  ];
  const mockSetTodos = jest.fn();

  beforeEach(() => {
    render(<CreateTodoPage todos={mockTodos} setTodos={mockSetTodos} />);
    mockSetTodos.mockClear();
  });

  it("should take a snapshot for CreateTodoPage", () => {
    const { asFragment } = render(
      <CreateTodoPage todos={mockTodos} setTodos={mockSetTodos} />
    );
    expect(
      asFragment(<CreateTodoPage todos={mockTodos} setTodos={mockSetTodos} />)
    ).toMatchSnapshot();
  });
  it("should have the required elements when CreateTodoPage is rendered", () => {
    expect(screen.findAllByText("SUBMIT")).toBeTruthy();
    expect(screen.findAllByText("HI, USER")).toBeTruthy();
  });
});
