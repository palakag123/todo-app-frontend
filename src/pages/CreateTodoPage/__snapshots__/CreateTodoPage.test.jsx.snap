// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CreateTodoPage should take a snapshot for CreateTodoPage 1`] = `
<DocumentFragment>
  <div
    class="create-todo-page-container"
  >
    <div
      class="details-container"
    >
      <div
        class="details-inner-container"
      >
        <img
          alt="user"
          data-testid="userImg"
          id="user-img"
          src="userImage.jpeg"
        />
        <h2
          data-testid="greeting"
          id="greeting"
        >
          HI, USER
        </h2>
        <br />
        <h1
          data-testid="time"
          id="date-time"
        >
          <time
            datetime="1650831384901"
          >
            1:46 am
          </time>
        </h1>
        <h1
          data-testid="date"
          id="date-time"
        >
          <time
            datetime="1650831384901"
          >
            25 April, 2022
          </time>
        </h1>
      </div>
    </div>
    <div
      class="create-todo-container"
    >
      <form
        class="new-todo-form"
        data-testid="newTodoForm"
      >
        <textarea
          cols="5"
          data-testid="newTodoInput"
          id="new-todo-input"
          maxlength="100"
          rows="5"
        />
        <div
          class="count-char-container"
        >
          <p
            data-testid="charCount"
            id="character-count"
          >
            100 characters left
          </p>
        </div>
        <button
          data-testid="submitButton"
          id="submit-button"
          type="submit"
        >
          SUBMIT
        </button>
      </form>
      <div
        class="back-container"
        data-testid="backArrow"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
          data-testid="backArrowButton"
          focusable="false"
          id="back-arrow"
          viewBox="0 0 24 24"
        >
          <path
            d="M20 11H7.83l5.59-5.59L12 4l-8 8 8 8 1.41-1.41L7.83 13H20v-2z"
          />
        </svg>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
